src/utils/helpers.js

```diff
@@ -0,0 +1,7 @@
+export function setButtontext(btn, isLoading, loadingText = "Saving...", defaultText = "Save") {
+    if (isLoading) {
+        btn.textContent = isLoading ? loadingText : defaultText;
+    } else {
+        btn.disabled = isLoading; 
```

Comment:
Please, delete ` btn.disabled = isLoading; ` because it makes problems for the button text changing. And it will make problems for the button disabling in `.then`

It should be like this:

```js
export function setButtontext(
  btn,
  isLoading,
  loadingText = "Saving...",
  defaultText = "Save"
) {
  btn.textContent = isLoading ? loadingText : defaultText;
}


```